crates/polars-plan/src/dsl/function_expr/range/int_range.rs

```diff
@@ -1,29 +1,62 @@
 use polars_core::prelude::*;
 use polars_core::series::{IsSorted, Series};
+use polars_core::with_match_physical_integer_polars_type;
 
 use super::utils::ensure_range_bounds_contain_exactly_one_value;
 
-pub(super) fn int_range(s: &[Series], step: i64) -> PolarsResult<Series> {
-    let start = &s[0];
-    let end = &s[1];
+pub(super) fn int_range(s: &[Series], step: i64, dtype: DataType) -> PolarsResult<Series> {
+    let mut start = &s[0];
+    let mut end = &s[1];
 
     ensure_range_bounds_contain_exactly_one_value(start, end)?;
 
-    match start.dtype() {
-        dt if dt == &IDX_DTYPE => {
-            let start = start.idx()?.get(0).unwrap();
-            let end = end.cast(&IDX_DTYPE)?;
-            let end = end.idx()?.get(0).unwrap();
-            int_range_impl::<IdxType>(start, end, step)
-        },
-        _ => {
-            let start = start.cast(&DataType::Int64)?;
-            let end = end.cast(&DataType::Int64)?;
-            let start = start.i64()?.get(0).unwrap();
-            let end = end.i64()?.get(0).unwrap();
-            int_range_impl::<Int64Type>(start, end, step)
-        },
+    let (start_storage, end_storage);
+    if *start.dtype() != dtype {
+        start_storage = start.cast(&dtype)?;
+        start = &start_storage;
     }
+    if *end.dtype() != dtype {
+        end_storage = end.cast(&dtype)?;
+        end = &end_storage;
+    }
+
+    with_match_physical_integer_polars_type!(dtype, |$T| {
+        let start_ca: &ChunkedArray<$T> = start.as_any().downcast_ref().unwrap();
+        let end_ca: &ChunkedArray<$T> = end.as_any().downcast_ref().unwrap();
+        let start_v = start_ca.get(0).unwrap();
+        let end_v = end_ca.get(0).unwrap();
+        int_range_impl::<$T>(start_v, end_v, step)
+    })
+}
+
+fn int_range_impl<T>(start: T::Native, end: T::Native, step: i64) -> PolarsResult<Series>
```

Comment:
This function stayed exactly the same, but I moved it from further down in the file.

Reply:
Nicely abstracted. :+1: